include/oneapi/dpl/internal/dynamic_selection_impl/fixed_resource_policy.h

```diff
@@ -45,33 +45,32 @@ namespace experimental {
 
     struct state_t {
         resource_container_t resources_;
-        int offset_;
+        size_t offset_;
     };
 
     std::shared_ptr<state_t> state_;
 
   public:
-    fixed_resource_policy(int offset=0) {
-        if(offset!=deferred_initialization){
-            initialize(offset);
-        }
+    fixed_resource_policy(size_t offset=0) {
+        initialize(offset);
     }
 
-    fixed_resource_policy(const std::vector<resource_type>& u, int offset=0) {
-        if(offset!=deferred_initialization){
-            initialize(u, offset);
-        }
+    fixed_resource_policy(deferred_initialization_t) {}
+
+
+    fixed_resource_policy(const std::vector<resource_type>& u,  size_t offset=0) {
```

Comment:
Here and in all other places better to use ```::std::size_t``` like in other oneDPL code

Reply:
Fixed